dotCMS/src/main/java/com/dotcms/ai/rest/SearchResource.java

```diff
@@ -203,10 +205,13 @@ public final Response relatedByPost(@Context final HttpServletRequest request,
                                     user,
                                     true);
 
+        if (!UtilMethods.isSet(contentlet)) {
+            Logger.warn(this.getClass(), getFailMessage(identifier, inode, language));
+            return Response.status(404).build();
```

Comment:
the approach now is throw exception for these cases and view instead of a response